test/test_grid.py

```diff
@@ -96,15 +96,15 @@ def test_grid_with_holes():
     assert grid_without_holes.global_sphere_coverage
 
 
-def test_grid_encode_as():
+def test_grid_to_xarray():
     """Reads a ugrid file and encodes it as `xarray.Dataset` in various types."""
-    grid_CSne30.encode_as("UGRID")
-    grid_RLL1deg.encode_as("UGRID")
-    grid_RLL10deg_CSne4.encode_as("UGRID")
+    grid_CSne30.to_xarray("UGRID")
+    grid_RLL1deg.to_xarray("UGRID")
+    grid_RLL10deg_CSne4.to_xarray("UGRID")
 
-    grid_CSne30.encode_as("Exodus")
-    grid_RLL1deg.encode_as("Exodus")
-    grid_RLL10deg_CSne4.encode_as("Exodus")
+    grid_CSne30.to_xarray("Exodus")
+    grid_RLL1deg.to_xarray("Exodus")
+    grid_RLL10deg_CSne4.to_xarray("Exodus")
```

Comment:
For all `to_xarray()` tests, we need to also add a step that reads in the saved output and checks to make sure the output is valid.

Right now, these tests only check if the function can be executed, which may not be sufficient.

Reply:
thanks for this comment, found a few bugs in the readers. Fixes pushed.